### PR TITLE
Bump SDK version numbers to 2.0 where appropriate

### PR DESCRIPTION
**Pull request by bame-da**
CHANGELOG_BEGIN
CHANGELOG_END

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [x] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [x] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [x] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description
- [ ] If you mean to change the status of a component, please make sure you keep [the Component Status page](https://github.com/digital-asset/daml/blob/main/docs/source/support/component-statuses.rst) up to date.

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: docs/source/json-api/index.rst [hunk 1]
```diff
@@ -200,9 +200,9 @@ start the standalone JAR, you can use the following command:
 
 .. code-block:: shell
 
-    java -jar http-json-1.5.0.jar --config json-api-app.conf
+    java -jar http-json-2.0.0.jar --config json-api-app.conf
 
-Replace the version number ``1.5.0`` by the version of the SDK you are
+Replace the version number ``2.0.0`` by the version of the SDK you are
```
Comment (stefanobaghino-da) [inline docs/source/json-api/index.rst:205]: This is bound to become obsolete very fast. I would rather recommend to point at a generic `<version>` or something along those lines.

File: docs/source/ops/connect/auth0.rst [hunk 2]
```diff
@@ -58,15 +58,11 @@ In order to follow along this guide, you will need:
...
-- To be running SDK 1.17.0 or later. Before 1.17.0, the JSON API required an
-  extra token, the setup of which is not covered here. If you are somehow
-  unable to upgrade but still want to use Auth0, please contact us for
-  assistance.
 - An application you want to deploy on your Daml system. This is not, strictly
   speaking, required, but the whole experience is going to be a lot less
   satisfying if you don't end up with something actually running on your Daml
   system. In this guide, we'll use the `create-daml-app` template,
-  which as of Daml SDK 1.17.0 supports Auth0 out-of-the-box on its UI side.
+  which supports Auth0 out-of-the-box on its UI side.
```
Comment (stefanobaghino-da) [inline docs/source/ops/connect/auth0.rst:65]: Feel free to change this, but we'll soon remove this page as it heavily depends on the Helm Chart (temporary measure until the Helm Chart is restored).

File: docs/source/ops/connect/helm.rst [hunk 3]
```diff
@@ -6,7 +6,7 @@
 Daml Helm Chart
 ===============
 
-As of 1.18.0, we provide an Early Access version of the Helm Chart for
+We provide an Early Access version of the Helm Chart for
```
Comment (stefanobaghino-da) [inline docs/source/ops/connect/helm.rst:9]: Feel free to change this, but we'll soon remove this page (temporary measure until the Helm Chart is restored).

File: docs/source/tools/assistant.rst [hunk 4]
```diff
@@ -224,9 +224,9 @@ To install the SDK version specified in the project config, run::
 
   daml install project
 
-To install a specific SDK version, for example version ``1.7.0``, run::
+To install a specific SDK version, for example version ``2.0.0``, run::
 
-  daml install 1.7.0
+  daml install 2.0.0
```
Comment (stefanobaghino-da) [inline docs/source/tools/assistant.rst:229]: Again, I would recommend pointing at a generic `<version>`.

File: docs/source/tools/codegen.rst [hunk 5]
```diff
@@ -60,7 +60,7 @@ this feature.
 
 Here is an example::
 
-    sdk-version: 1.2.0
+    sdk-version: 2.0.0
```
Comment (stefanobaghino-da) [inline docs/source/tools/codegen.rst:63]: Same here.

**Review by stefanobaghino-da — COMMENTED** [review]
I don't think that pointing to a specific version is particularly wise. I would rather recommend putting a placeholder that users can recognize. Better to have something that fails for easily recognizable reasons, then something that we need to keep up to date or risk having weird failures due to random copy-pasting.

In the future we may want to evaluate some way to replace the placeholder with the current version, but definitely not needed right now.

**Comment by bame-da** [comment]
@stefanobaghino-da I see your concerns, but because these are all "examples", I do think they should be working code, even if obsolete fast. We can bump versions again with 3.0.

**Review by cocreature — APPROVED** [review]
seems better than the previous state so given the timelines happy to merge it.

File: docs/source/ops/connect/auth0.rst (hunk 2 above)
Comment (garyverhaegen-da) [reply docs/source/ops/connect/auth0.rst:65]: The `create-daml-app` support for Auth0 predates, and does not depend on, the Helm chart. That said, I'm not claiming it still works in 2.0.

**Pull request merged**